compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -1524,13 +1540,27 @@ TR::X86RegImmInstruction::addMetaDataForCodeAddress(uint8_t *cursor)
                setReloKind(TR_RamMethod);
             // intentional fall-through
          case TR_RamMethod:
+            symbolKind = TR::SymbolType::typeMethod;
             // intentional fall-through
          case TR_ClassPointer:
-            cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor,
-                                                              (uint8_t*)getNode(),
-                                                              (TR_ExternalRelocationTargetKind) _reloKind,
-                                                              cg()),
-                                   __FILE__, __LINE__, getNode());
+            if (symbolKind == TR::SymbolType::typeClass && cg()->comp()->getOption(TR_UseSymbolValidationManager))
```

Comment:
why only typeClass? are you intentionally not handling typeMethod with the SymbolManager?

Reply:
Same as above.